lib/config/configuration.go

```diff
@@ -2173,24 +2231,31 @@ func splitRoles(roles string) []string {
 
 // applyTokenConfig applies the auth_token and join_params to the config
 func applyTokenConfig(fc *FileConfig, cfg *service.Config) error {
-	if fc.AuthToken != "" {
-		cfg.JoinMethod = types.JoinMethodToken
-		cfg.SetToken(fc.AuthToken)
-	}
+	// config v1 and v2 allow for `auth_token` to set `join_params.method` and `join_params.token_name`
+	if fc.Version == defaults.TeleportConfigVersionV1 || fc.Version == defaults.TeleportConfigVersionV2 {
+		if fc.AuthToken != "" {
+			if fc.JoinParams != (JoinParams{}) {
+				return trace.BadParameter("only one of auth_token or join_params should be set")
+			}
 
-	if fc.JoinParams != (JoinParams{}) {
-		if cfg.HasToken() {
-			return trace.BadParameter("only one of auth_token or join_params should be set")
+			cfg.JoinMethod = types.JoinMethodToken
+			cfg.SetToken(fc.AuthToken)
+
+			return nil
 		}
+	} else if fc.AuthToken != "" {
+		// from config v3 onwards, we have removed auth_token
+		return trace.BadParameter("auth_token is no longer supported in config v3 onwards, use join_params instead")
```

Comment:
I do dislike `token_name` being used for the value of auth token, as it can be a file path or an actual token, not the name of a token.

However, I agree that `token_name` makes sense when the `method` isn't `token`.

Reply:
I'm torn here. While I agree that `token_name` is an unfortunate key for something that can also be the token value, I'm with Noah in that I prefer when there's not several different ways to configure the same thing.

That said, sometimes the easiest thing to do is to simply do nothing. We don't need to include join params in scope for the v3 config change, I just thought it would be nice to solve that while we're introducing a new config version.

Reply:
We ended up leaving `auth_token` alone and removed it from the scope of config v3.